### ISSUE TITLE
Safari < 8.0 doesn't display the canvas

### ISSUE THREAD

**Issue by quentinlechemia** [imgly/imgly-sdk-html5#61]
Hi,

we had a lot of users who tried to upload their picture from an iPhone 5s (or 4s, 5...) or a Mac. They couldn't see the imgly canvas. We figured out that your main script is bugging out on Safari < 8.0. Do you have any solution to fix it? Thank you so much.

Cheers,

Quentin

**Comment by quentinlechemia** [comment]
I found the problem. Safari generates "Can't find variable" error because there are no promises for iOS Safari 7.1, or Safari 7. (http://stackoverflow.com/questions/22462396/cant-find-variable-promise-on-safari)
I solved this by adding the RVSP.js library (http://www.toptal.com/javascript/javascript-promises) and replace all the "Promise" variables by "RSVP.Promise". 
I guess there is a better solution but for the moment, it works!... Cheers.
